### ISSUE TITLE
plan9 build broken: Make:  Don't know how to make TESTFRAMEWORK=echo.  Stop.

### ISSUE THREAD

**Issue by mischief** [Orc/discount#192]
here's a current build log of 2.2.4 plus #191's fix:

```
cpu% mk config
ape/psh -c 'cd .. && ./configure.sh $CONFIG'
Configuring for [markdown]
checking the C compiler (cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION) ok
Looking for cpp (using $CC -E as a cpp pipeline)
looking for install (/bin/install)
(install -s does not appear to work?)
Checking __attribute__((__destructor__)) (no)
looking for ar (/bin/ar)
looking for ranlib (/bin/ranlib)
looking for pkg-config (not found)
looking for automake (not found)
checking for "volatile" keyword (found)
checking for "const" keyword (found)
Checking for "inline" keyword (found)
defining WORD & DWORD scalar types
looking for a reentrant basename (not found)
looking for header stdlib.h (found)
looking for the alloca function (not found)
looking for header sys/types.h (found)
looking for header pwd.h (found)
looking for the getpwuid function (found)
looking for header sys/stat.h (found)
looking for the stat function (found)
special file macros in sys/stat.h:none.
looking for the srandom function (not found)
looking for the srand function (found)
looking for the memset function (found)
looking for the random function (not found)
looking for the rand function (found)
looking for the strcasecmp function (found)
looking for the strncasecmp function (found)
looking for the fchdir function (not found)
looking for the getcwd function (found)
looking for header malloc.h (not found)
looking for find (not found)
looking for "ln -s" (/bin/ln exists, but -s does not seem to work)
looking for ln (/bin/ln)
looking for ar (/bin/ar)
looking for ranlib (/bin/ranlib)
looking for sed (/bin/sed)
generating Makefile
generating version.c
generating mkdio.h
cpu% mk markdown
ape/psh -c 'cd .. && $CC mktags.c -o mktags && ./mktags > blocktags && make'
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -I. -c main.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -I. -c pgm_options.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -I. -c gethopt.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -I. -c notspecial.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c mkdio.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c markdown.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c dumptree.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c generate.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c resource.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c docheader.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -o branch tools/branch.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -DBRANCH=`./branch` -DVERSION=\"`cat VERSION`\" -c version.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c toc.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c css.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c xml.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c Csio.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c xmlpage.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c basename.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c emmatch.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c github_flavoured.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c setup.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c tags.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c html5.c
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -g -c flags.c
Make:  Don't know how to make TESTFRAMEWORK=echo.  Stop.
mk: ape/psh -c 'cd ...  : exit status=rc 2599042: psh 2599044: sh 2599049: 1
```

i have no clue what's wrong with Make. all of the targets in `$TESTFRAMEWORK` build fine on their own.

**Comment by Orc** [comment]
What happens if you just do `make TESTFRAMEWORK=echo`?  Does it still fail for you?

**Comment by mischief** [comment]
yes:

```
$ make
Make:  Don't know how to make TESTFRAMEWORK=echo.  Stop.
$ make TESTFRAMEWORK=echo
Make:  Don't know how to make TESTFRAMEWORK=echo.  Stop.
$ make 'TESTFRAMEWORK=echo'
Make:  Don't know how to make TESTFRAMEWORK=echo.  Stop.
$ make "TESTFRAMEWORK=echo"
Make:  Don't know how to make TESTFRAMEWORK=echo.  Stop.
$ make echo
cc -D_BSD_EXTENSION -D_C99_SNPRINTF_EXTENSION -I. -o echo tools/echo.c
```

make seems to think `TESTFRAMEWORK=echo` is a target name when evaluating all's dependencies :/

**Comment by Orc** [comment]
Ugh.  Maybe TESTFRAMEWORK is too long of an identifier?  Dumb test; go into the makefile and change all TESTFRAMEWORK to THINGS , then see if it still blows up   (OBJS is defined in the same way that TESTFRAMEWORK is, so if it was a matter of the plan9 unix environment's make not liking macros in the makefile you'd not get even that far.)

**Comment by Orc** [comment]
(And which version of plan9 are you running?)

**Comment by mischief** [comment]
ok. i think i found the issue.. looks like a make parser bug.

```
$ make -p
Macros:
	PKGDIR = (null)
	DESTDIR = (null)
	TESTFRAMEWORK = (null)
	MAN3PAGES = mkd-callbacks.3 mkd-functions.3 markdown.3 mkd-line.3
	COMMON = pgm_options.o gethopt.o notspecial.o
	OBJS = mkdio.o markdown.o dumptree.o generate.o  resource.o docheader.o version.o toc.o css.o  xml.o Csio.o xmlpage.o basename.o emmatch.o  github_flavoured.o setup.o tags.o html5.o flags.o  TESTFRAMEWORK=echo cols branch pandoc_headers
	MKDLIB = libmarkdown
	SAMPLE_PGMS+ = theme
	SAMPLE_PGMS = mkd2html makepage
	PGMS = markdown
	INCDIR = /usr/local/include
	LIBDIR = /usr/local/lib
	MANDIR = /usr/local/man
	BINDIR = /usr/local/bin
	RANLIB = /bin/ranlib
```

here we can see TESTFRAMEWORK is (null) and TESTFRAMEWORK within OBJS.

generated Makefile snippet:

```
OBJS=mkdio.o markdown.o dumptree.o generate.o \
     resource.o docheader.o version.o toc.o css.o \
     xml.o Csio.o xmlpage.o basename.o emmatch.o \
     github_flavoured.o setup.o tags.o html5.o flags.o \
      
TESTFRAMEWORK=echo cols branch pandoc_headers
```

@AMALLOC@ @H1TITLE@ were null when generating Makefile it seems, but the line continuation persist somehow making the parser eat TESTFRAMEWORK.

**Comment by mischief** [comment]
this is 9front.

**Comment by Orc** [comment]
Okay, as OS bugs that's one that might be easy to work around;  I rearranged the order of file in the OBJS= macro to have flags.o be the very last one after the @AMALLOC@ & @H1TITLE@, so unless the plan9 make has completely lost its mind it should find text on that continuation line and not continue to continue.

**Comment by mischief** [comment]
in my local change, i simply deleted the trailing `\` and it worked. but there's more errors..! :-)

**Comment by mischief** [comment]
we can probably fix 9front's make to parse this correctly, but i do not think plan 9 from bell labs (or any of the other distributions) will be fixed any time soon. so changing it will benefit users of other distributions, as silly as that is.

i'll report the new error in a new issue.

fixed in 11c757d70b4123289666fe76fd452d3453489a66
